File: apps/mobile/src/native/StudienbuchInlineModule.ts
import { NativeModule, requireNativeModule } from "expo";

declare class StudienbuchInlineModule extends NativeModule {
  moduleKind: "inline";
  getModuleBoundary(): string;
  getPlatformSummary(): string;
}

export default requireNativeModule<StudienbuchInlineModule>("StudienbuchInlineModule");

File: apps/mobile/src/native/StudienbuchInlineModule.kt
package dev.schnau.studienbuch.inline

import android.os.Build
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class StudienbuchInlineModule : Module() {
  override fun definition() = ModuleDefinition {
    Constant("moduleKind") {
      "inline"
    }

    Function("getModuleBoundary") {
      "apps/mobile/src/native"
    }

    Function("getPlatformSummary") {
      "Android ${Build.VERSION.RELEASE} (API ${Build.VERSION.SDK_INT}) via Kotlin inline module"
    }
  }
}
